### PR TITLE
Ms1 list structure

### PR DESCRIPTION
**Pull request by Bishoy-Samwel**
- [x] Set up a new project with webpack that is based on the webpack exercise you have already completed.
- [x]  Create an index.html file and write your HTML markup here.
- [x]  Create an empty To-Do List placeholder (div or ul element).
- [x]  The index.html file must be set as a template using the HTML Webpack Plugin.
- [x]  Create an index.js file and
- [x]  set an array of some simple to-do tasks (array of objects).
- [x]  Each task object should contain three keys:
- [x]  Write a function to iterate over the tasks array and populate an HTML list item element for each task.
- [x]  On page load renders the dynamically created list of tasks in the dedicated placeholder.
- [x]  The list should appear in order of the index values for each task.
- [x]  Create a style.css and set rules for the To-Do List. CSS must be loaded by Webpack Style/CSS Loader.
- [x]  Your list should look like the part of the minimalist project captured in the video below.
- [x]  All your source files (index.html, index.js, and style.css) must be located in /src directory, and    *  your distribution files will be generated by webpack and served by webpack dev server from /dist folder.

### CODE REVIEW

File: src/list.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  constructor() {
+    this.tasks = JSON.parse(localStorage.getItem('tasks')) || List.initialize();
+  }
+
+  static initialize() {
+    const task1 = new Task('Eat', 0);
+    const task2 = new Task('Rest', 1);
+    const task3 = new Task('Sleep', 2);
+    return [task1, task2, task3];
+  }
```
Comment (hemantso) [inline src/list.js:13]: - [x]  As per project requirements each task object should contain three keys. I can see there is one key missing like for `completed` that should be a boolean datatype. Kindly try to fix this.

File: src/print.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default function printMe() {
+  console.log('I get called from print.js!');
+}
```
Comment (hemantso) [inline src/print.js]: - [x] Kindly try to delete the console.log statements. It is not a good practice to push it with code.

File: src/list.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+const list = new List();
+console.log(list.tasks);
```
Comment (hemantso) [inline src/list.js]: - [x] Kindly try to delete the console.log statements. It is not a good practice to push it with code.

**Review by hemantso — CHANGES_REQUESTED** [review]
Hi @Bishoy-Samwel , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there! 

Alt Text
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Kindly try to add a descriptive `README` file with the information to run the project.

Cheers,
Hemant Soni
Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/list.js (hunk 1 above)
Comment (Bishoy-Samwel) [reply src/list.js:13]: @hemant-soni-vst-au4 Have you seen the constructor?

>   constructor(desc, index) {
    this.desc = desc;
    this.completed = false;
    this.index = index;
    this.id = Task.generateID();
  }

Comment (hemantso) [reply src/list.js:13]: - Yes, I just noticed it. Everything looks perfect for now. you are good to go. Just raise a CR request.
Happy Coding.

**Review by mikenath223 — APPROVED** [review]
Hello there  

Your project is complete!  There is nothing else to say other than... it's time to merge it 🛳 
Congratulations! 🎉 

Happy coding. Cheers! 🚀🚀

If you feel something isn't clear enough you can leave a question or comment in the PR thread.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**